File: src/main/java/com/store/config/SecurityCookieProperties.java
package com.store.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "app.security.cookie")
@Data
public class SecurityCookieProperties {

    private boolean secure;

    private String sameSite = "Lax";

    private String path = "/api/account";

    private long maxAgeSeconds;
}

File: src/main/java/com/store/config/SecurityCookieProperties.kt
package com.store.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app.security.cookie")
data class SecurityCookieProperties(
    val isSecure: Boolean = false,
    val sameSite: String = "Lax",
    val path: String = "/api/account",
    val maxAgeSeconds: Long = 0,
)
